src/main/java/org/spongepowered/common/entity/SpongeEntityArchetype.java

```diff
@@ -0,0 +1,145 @@
+
+import java.util.Arrays;
+
+public class SpongeEntityArchetype extends AbstractArchetype<EntityType, EntitySnapshot> implements EntityArchetype {
+
+    final EntityType entityType;
+
+    SpongeEntityArchetype(SpongeEntityArchetypeBuilder builder) {
+        super(builder.entityType, NbtTranslator.getInstance().translateData(builder.entityData));
+        this.entityType = builder.entityType;
+    }
+
+    @Override
+    public EntityType getType() {
+        return this.entityType;
+    }
+
+    @Override
+    public DataContainer getEntityData() {
+        return NbtTranslator.getInstance().translateFrom(this.data);
+    }
+
+    @Override
+    public boolean apply(Location<World> location, Cause cause) {
+        final NBTTagCompound data = (NBTTagCompound) this.data.copy();
+        final Vector3d position = location.getPosition();
+        final double x = position.getX();
+        final double y = position.getY();
+        final double z = position.getZ();
+        final BlockPos blockPos = new BlockPos(x, y, z);
+        final World world = location.getExtent();
+        final WorldServer worldServer = (WorldServer) world;
+
+        final Entity entity = EntityList.createEntityFromNBT(data, worldServer);
+
+        final org.spongepowered.api.entity.Entity spongeEntity = EntityUtil.fromNative(entity);
+        final SpawnEntityEvent.Custom event = SpongeEventFactory.createSpawnEntityEventCustom(cause, Arrays.asList(spongeEntity), world);
+        if (!event.isCancelled()) {
+            final IMixinWorldServer mixinWorldServer = (IMixinWorldServer) worldServer;
+            entity.setPositionAndRotation(x, y, z, entity.rotationYaw, entity.rotationPitch);
+            if (entity instanceof EntityLiving) {
+                mixinWorldServer.forceSpawnEntity(entity);
+                ((EntityLiving) entity).onInitialSpawn(worldServer.getDifficultyForLocation(blockPos), null);
+                ((EntityLiving) entity).spawnExplosionParticle();
+            } else {
+                mixinWorldServer.forceSpawnEntity(entity);
+            }
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public EntitySnapshot toSnapshot(Location<World> location) {
+        final SpongeEntitySnapshotBuilder builder = new SpongeEntitySnapshotBuilder();
+        builder.entityType = this.entityType;
+        builder.compound = (NBTTagCompound) this.data.copy();
+        builder.worldId = location.getExtent().getUniqueId();
+        builder.position = location.getPosition();
+        return builder.build();
+    }
+
+    @Override
+    public int getContentVersion() {
+        return DataVersions.EntityArchetype.BASE_VERSION;
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return new MemoryDataContainer()
+                .set(DataQueries.EntityArchetype.ENTITY_TYPE, this.type)
+                .set(DataQueries.EntityArchetype.ENTITY_DATA, this.data)
+                ;
+    }
```

Comment:
Can we get a standard way of serialising to MC format with EntityType within the data. If not I'll add it to my PR for mob spawners which use archetypes in place of located snapshots.